scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/StaxJobFactory.java

```diff
@@ -167,6 +168,27 @@ public Job createJob(URI filePath, Map<String, String> replacementVariables,
         }
     }
 
+    @Override
+    public Job createJob(InputStream workflowStream) throws JobCreationException {
+        return createJob(workflowStream, null, null);
+    }
+
+    @Override
+    public Job createJob(InputStream workflowStream, Map<String, String> replacementVariables,
+            Map<String, String> replacementGenericInfos) throws JobCreationException {
+        try {
+            File temporaryJobContentFile = File.createTempFile("jobContent" +
+                                                               workflowStream.toString().replaceAll("\\W+", ""),
+                                                               ".tmp");
+            FileUtils.copyInputStreamToFile(workflowStream, temporaryJobContentFile);
+            return createJob(temporaryJobContentFile, replacementVariables, replacementGenericInfos);
```

Comment:
Maybe you  want to add finally with temporaryJobContentFile.delete()

Reply:
Indeed it seems crucial to me. Thank you.